layer/training/runtime/model_trainer.py

```diff
@@ -38,7 +46,71 @@ class TrainContextDataclassMixin:
     train_index: Optional[str] = None
 
 
+cpu_used_temp = None  # pylint: disable=C0103
```

Comment:
why do you define this as a global var? same for `start_time_temp`

Reply:
I think a better way would be to create a separate class holding these values, then attach an instance of this class somewhere, such as the train context. Then in your thread you pass this object as an argument, so that you can avoid working with globals

Reply:
About the step value, the logic is a bit not straightforward to grasp as an idea. Are you somehow trying to get the around the problem of us displaying metrics with step on the x axis, rather than time and emulating that through increasing the step value?
And if that's the case, this is kinda problematic because the step value on this graph would not be in sync with the ones on the other graphs. And worse, the step is generally intended to mean a specific intermediate state of the training of an ML model while here it would mean something else.

Reply:
>I think a better way would be to create a separate class holding these values, then attach an instance of this class somewhere, such as the train context. Then in your thread you pass this object as an argument, so that you can avoid working with globals

I can look into doing that, it might take me a little while however.

Reply:
>About the step value, the logic is a bit not straightforward to grasp as an idea. Are you somehow trying to get the around the problem of us displaying metrics with step on the x axis, rather than time and emulating that through increasing the step value?
And if that's the case, this is kinda problematic because the step value on this graph would not be in sync with the ones on the other graphs. And worse, the step is generally intended to mean a specific intermediate state of the training of an ML model while here it would mean something else.

It's true that right now the step annotations might make this confusing, however we're also planning to change how the graphs are displayed, which should address this.
Using steps is the only way to display a graph for now, so that's what we end up using.